lib/rule.ml

```diff
@@ -55,38 +56,42 @@ module Prefix = struct
       allow list is considered a prefix match of length 0. The ignore list is
       evaluated before the allow list. *)
   let match_rules filename ~rules =
-    let compare a b = Int.compare (snd a) (snd b) in
-    let is_prefix prefix = String.is_prefix filename ~prefix in
+    let max_elt t =
+      let compare a b = Int.compare (snd a) (snd b) in
+      match t with
+      | [] -> None
+      | v :: vs -> Some (List.fold_left (fun a b -> if compare b a > 0 then b else a) v vs) in
+    let is_prefix prefix = Stre.starts_with filename prefix in
```

Comment:
One thing I noticed while doing this is that in `monorobot.opam` the minimum required version for ocaml is `>=4.08.0`. There are some API changes in later versions that would make life easier in some places, like `String.starts_with` (here) and `String.fold_left` (`lib/mrkdwn.ml` line 11). Right now they're replaced with equivalents from `Extlib` and `Devkit`.  Should we bump the minimum version up to 4.14? I think that's the compiler installed by default when running `opam switch create ./` anyway.

Reply:
Yes, let's do that. Thanks!